src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java

```diff
@@ -667,6 +667,152 @@ else if (useFilteredViolations) {
         }
     }
 
+    public static List<TestInputViolation> getViolationsForGoogleConfig(String inputFilePath)
+            throws Exception {
+        final Path filePath = Paths.get(inputFilePath);
+        final List<String> lines = readFile(filePath);
+        final List<TestInputViolation> violations = new ArrayList<>();
+        for (int lineNo = 0; lineNo < lines.size(); lineNo++) {
+            violations.addAll(getViolationsForGoogleConfig(lines,
+                    lineNo));
+        }
+
+        return violations;
+    }
+
+    /**
+     * Gets the violations based on the lines provided for Google config.
+     *
+     * @param lines all the lines in the file.
+     * @param lineNo the current line number.
+     * @return the list of violations.
+     * @noinspection IfStatementWithTooManyBranches
+     * @noinspectionreason IfStatementWithTooManyBranches - complex logic of violation
+     *      parser requires giant if/else
+     * @throws CheckstyleException if violation message is not specified.
+     */
+    // -@cs[ExecutableStatementCount] splitting this method is not reasonable.
+    // -@cs[JavaNCSS] splitting this method is not reasonable.
+    // -@cs[CyclomaticComplexity] splitting this method is not reasonable.
+    private static List<TestInputViolation> getViolationsForGoogleConfig(List<String> lines,
+         int lineNo) throws CheckstyleException {
```

Comment:
copy of this is concern.

can we reuse:
```
private static void setViolations(TestInputConfiguration.Builder inputConfigBuilder,
                                      List<String> lines, boolean useFilteredViolations,
                                      int lineNo, boolean specifyViolationMessage)
            throws CheckstyleException {
```

you use useFilteredViolations as true
and we just need to go inputConfigBuilder.getViolations to get only list of violations.
In some future we will refactor this code at bit better.

Reply:
Done. Thanks for the advice, this helped a lot to reduce redundant code